cfgmgr/coppmgr.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "dbconnector.h"
+#include "orch.h"
+#include "producerstatetable.h"
+#include <unistd.h>
+
+#include <map>
+#include <set>
+#include <string>
+
+namespace swss {
+
+/* COPP Trap Table Fields */
+#define COPP_TRAP_ID_LIST_FIELD                "trap_ids"
+#define COPP_TRAP_GROUP_FIELD                  "trap_group"
+
+/* COPP Group Table Fields */
+#define COPP_GROUP_QUEUE_FIELD                 "queue"
+#define COPP_GROUP_TRAP_ACTION_FIELD           "trap_action"
+#define COPP_GROUP_TRAP_PRIORITY_FIELD         "trap_priority"
+#define COPP_GROUP_POLICER_METER_TYPE_FIELD    "meter_type"
+#define COPP_GROUP_POLICER_MODE_FIELD          "mode"
+#define COPP_GROUP_POLICER_COLOR_FIELD         "color"
+#define COPP_GROUP_POLICER_CBS_FIELD           "cbs"
+#define COPP_GROUP_POLICER_CIR_FIELD           "cir"
+#define COPP_GROUP_POLICER_PBS_FIELD           "pbs"
+#define COPP_GROUP_POLICER_PIR_FIELD           "pir"
+#define COPP_GROUP_POLICER_ACTION_GREEN_FIELD  "green_action"
+#define COPP_GROUP_POLICER_ACTION_RED_FIELD    "red_action"
+#define COPP_GROUP_POLICER_ACTION_YELLOW_FIELD "yellow_action"
+
+/* sflow genetlink fields */
+#define COPP_GROUP_GENETLINK_NAME_FIELD        "genetlink_name"
+#define COPP_GROUP_GENETLINK_MCGRP_NAME_FIELD  "genetlink_mcgrp_name"
+
+#define COPP_TRAP_TYPE_SAMPLEPACKET            "sample_packet"
+
+#define COPP_INIT_FILE "/etc/sonic/copp_cfg.json"
+
+struct CoppTrapConf
+{
+    std::string         trap_ids;
+    std::string         trap_group;
+};
+
+/* TrapName to TrapConf map  */
+typedef std::map<std::string, CoppTrapConf> CoppTrapConfMap;
+
+/* TrapGroupName to GroupConf map  */
```

Comment:
Is it TrapGroupName or TrapName?

Reply:
Modifying to /* TrapID to Trap group name map  */